inst/extdata/mvp/product-carbon-transition-risk_review.Rmd

```diff
@@ -0,0 +1,308 @@
+---
+title: "Product Carbon Transition Risk"
+author: "Tilman Trompke"
+params:
+  input1: NULL
+  input2: NULL
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+### Introduction
+
+The Product Carbon Transition Risk measures the transition risk on the
+product-level of a company. The indicator is expressed in the percentage
+of all products that are at high risk, medium risk or low risk due to
+the products’ relative carbon footprint. The assessment is first
+performed on a product-level and can then be aggregated to the
+company-level.
+
+The Product Carbon Transition Risk measures the relative carbon
+footprint per product. As a default option each product will be compared
+to the carbon footprint of every other product. Alternatively, users can
+also choose to compare those products of the same type, so e.g.,
+comparing only raw materials with each other. Products with a higher
+carbon-footprint will also face a higher risk.
+
+After identifying each carbon footprint for one product, the products
+will be ranked according to their carbon footprint. Products in the
+highest percentile (≥70%) will be classified as high transition risk
+products. Products in the medium percentile (between ≥30% and \<70%)
+will be classified as medium transition risk products. Products in the
+lowest percentile (\<30%) will be classified as low transition risk
+products.
+
+After categorisation, we will aggregate all products from the same
+category and set them in relation to all products that the company
+produces. We derive the Product Carbon Transition Risks.
+
+Please note that carbon footprints, and emissions are used equivalently.
+Carbon footprint refers only on the emissions which occur at the
+production stage of the product and not the emissions from the inputs.
+The unit is CO2e in kg.
+
+The goal of the transition risk MVP is to create a first draft of how
+the transition risk indicator for products would be build in code in
+order to convert it into code production easily in the future. This MVP
+should provide the share of products with “low”, “medium”, or “high”
+relative production emissions per company.
+
+In this markdown file the aim is to prepare data so that they contain
+the following:
+
+- a column with production emissions
+- a column indicating the percentile relative to (i) all products with
+  same unit (ii) products in same sector (iii) products in same segment
+- a column indicating whether the product has “low”, “medium” or “high”
+  relative production emissions
+
+The product carbon transition risk indicator consists of 2 main steps:
+
+    - Step 1: Calculating the relative carbon footprint per product
+    - Step 2: Aggregating on the company-level
```

Comment:
This suggests we'll need at least two main functions but maybe more. It's better to have many small focused functions that a single big one that does it all. Smaller functions are easier to understand, document, and test. To achieve the complexity the process requires we can simply combine multiple small functions. In pseudo code it might look like this:

```r
emissions_data |>
  calculate_carbon_footprint() |>
  aggregate_carbon_footprint(companies_data)
```